source/client/factories_impl.cc

```diff
@@ -168,18 +170,39 @@ RequestSourceFactoryImpl::create(const Envoy::Upstream::ClusterManagerPtr& clust
   for (const auto& option_header : request_options.request_headers()) {
     setRequestHeader(*header, option_header.header().key(), option_header.header().value());
   }
-
   if (!options_.requestSource().empty()) {
     RELEASE_ASSERT(!service_cluster_name.empty(), "expected cluster name to be set");
     // We pass in options_.requestsPerSecond() as the header buffer length so the grpc client
     // will shoot for maintaining an amount of headers of at least one second.
     return std::make_unique<RemoteRequestSourceImpl>(cluster_manager, dispatcher, scope,
                                                      service_cluster_name, std::move(header),
                                                      options_.requestsPerSecond());
+  } else if (options_.requestSourcePluginConfig().has_value()) {
+    absl::StatusOr<RequestSourcePtr> plugin_or = LoadRequestSourcePlugin(
+        options_.requestSourcePluginConfig().value(), api_, std::move(header));
+    RELEASE_ASSERT(plugin_or.ok(),
```

Comment:
Maybe annotate with a `XXX_GCOVR_EXCL_XXX`. We don't care not every branch gets hit in tests, putting that in documents an explicit choice that we're ok with that. If it would ever start working, it would also help with correcting the reported coverage % (it seems to be ignored today).

Reply:
As discussed on slack, we could probably just append `// NOT_REACHED_GCOVR_EXCL_LINE` to the line itself, or use 
```c++.
// GCOVR_EXCL_START
... code to be ignored in coverage measurements ...
// GCOVR_EXCL_STOP
```